### ISSUE TITLE
Opacity

### ISSUE THREAD

**Issue by pymsoft** [connollyst/zkgooglecharts#2]
There is no way to indicate the opacity of each column to any of the charts.

```
function drawChart() {
      var data = google.visualization.arrayToDataTable([
        ['Year', 'Visitations', { role: 'style' } ],
        ['2010', 10, 'color: gray'],
        ['2010', 14, 'color: #76A7FA'],
        ['2020', 16, 'opacity: 0.2'],
        ['2040', 22, 'stroke-color: #703593; stroke-width: 4; fill-color: #C5A5CF'],
        ['2040', 28, 'stroke-color: #871B47; stroke-opacity: 0.6; stroke-width: 8; fill-color: #BC5679; fill-opacity: 0.2']
      ]);
```

[https://developers.google.com/chart/interactive/docs/gallery/columnchart#column-styles](https://developers.google.com/chart/interactive/docs/gallery/columnchart#column-styles)

Best regards

Pedro

**Comment by connollyst** [comment]
Hey Pedro,

I've added a new [`ColumnRole`](https://github.com/connollyst/zkgooglecharts/blob/column-roles/src/main/java/org/zkoss/google/charts/data/ColumnRole.java) that can be used like so:

```
data.addColumn(ColumnType.STRING, ColumnRole.TOOLTIP);
data.addRow("Eat", 2, "Hello world!");
```

This works fine.

However, `ColumnRole.STYLE` has no effect in my tests. Want to see if you can get it to work?

I added a `console.log(data)` to [GoogleChart.js](https://github.com/connollyst/zkgooglecharts/blob/column-roles/src/main/resources/web/js/zul/googlecharts/GoogleChart.js) which shows you what data structure we're passing to the library. It looks good to me, not sure why it's not taking:

```
{
"cols":[
    {"type":"string","label":"Task","id":"task"},
    {"type":"number","label":"Hours per Day","id":"hours"},
    {"type":"string","role":"tooltip"},{"type":"string","role":"style"}
],
"rows":[
    {"c":[{"v":"Work"},{"v":11},{"v":null},{"v":null}]},
    {"c":[{"v":"Eat"},{"v":2},{"v":"Hello world!"},{"v":"opacity: 0.2"}]},
    {"c":[{"v":"Commute"},{"v":2},{"v":null},{"v":"fill-color: #76A7FA"}]},
    {"c":[{"v":"Watch TV"},{"v":2},{"v":"Hello world!"},{"v":"color: #76A7FA"}]},
    {"c":[{"v":"Sleep"},{"v":7,"f":"7.000"},{"v":null},{"v":"opacity: 0.7"}]}
]
}
```

**Comment by pymsoft** [comment]
Hi Sean, 

It's incredible... I try to put the google code in fiddle and not work...

https://jsfiddle.net/api/post/library/pure/

I put this code...: https://developers.google.com/chart/interactive/docs/gallery/columnchart#column-styles

mmmm, I found a code of google thats work:

view-source:https://google-developers.appspot.com/chart/interactive/docs/gallery/columnchart_4ca2d9c7f69846067884e8474c34ce78.frame

and I put this code in fiddle and works...

https://jsfiddle.net/mtdkkpqg/

Best regards

Pedro

**Comment by connollyst** [comment]
Great, that helps.

The main difference I see here is that the working example uses `google.visualization.arrayToDataTable` but [we use the DataTable constructor](https://github.com/connollyst/zkgooglecharts/blob/column-roles/src/main/resources/web/js/zul/googlecharts/GoogleChart.js#L54), passing in the structured data above.

I modified your example to not use `google.visualization.arrayToDataTable`
https://jsfiddle.net/gr4a6u3s/1/
It actually works. This is good, it means _it is possible_ to specify opacity using the data structure we use in zkgooglecharts.

I'll keep looking..

**Comment by connollyst** [comment]
Good news, bad news. So the code I implemented this morning actually does work!.. just not for pie charts. Actually, it doesn't work for pie charts even in JS:

Bar Chart, works: https://jsfiddle.net/gr4a6u3s/1/
Pie Chart, nope: https://jsfiddle.net/gr4a6u3s/2/

Any thoughts?

**Comment by pymsoft** [comment]
It seems that there is no way to make it work for pie charts.

https://jsfiddle.net/ktn0nbfz/

Maybe pie charts don't support roles style.

**Comment by connollyst** [comment]
Yup, I think you're right. I know bar charts are given special love in Google Charts, e.g.: only bar/column charts support the "material" style.

I've gone ahead and released [v0.3.0](http://search.maven.org/#artifactdetails%7Corg.zkoss%7Czkgooglecharts%7C0.3.0%7Cjar) which includes this new `ColumnRole` and you can use it to set opacity for bar & column charts.

#3 

Hope that does the trick. Thanks for the feedback, would love to hear any more suggestions
